rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -2026,8 +2029,9 @@ impl StructuralPageScheduler for FullZipScheduler {
         &'a mut self,
         io: &Arc<dyn EncodingsIo>,
     ) -> BoxFuture<'a, Result<Arc<dyn CachedPageData>>> {
-        // Check if we have a repetition index
-        if let Some(rep_index) = &self.rep_index {
+        // Check if caching is enabled and we have a repetition index
+        if self.enable_cache && self.rep_index.is_some() {
```

Comment:
Why add `self.enable_cache` here?  If it is false won't `self.rep_index.is_some` be false as well?